app/invocation/invocation_action_card.tsx

```diff
@@ -316,13 +318,28 @@ export default class InvocationActionCardComponent extends React.Component<Props
       })
       .catch((e) => {
         const error = BuildBuddyError.parse(e);
-        if (error.code === "NotFound") {
+        if (error.code === "NotFound" || error.code === "Unimplemented") {
           return;
         }
         errorService.handleError(e);
       })
       .finally(() => {
-        if (!executionFound && streamFallback) {
+        if (executionFound) return;
+
+        const fetchResult = this.fetchActionResult();
+        if (fetchResult !== null) {
+          fetchResult.catch((e) => {
+            const error = BuildBuddyError.parse(e);
+            if (error.code !== "NotFound") {
+              console.error("Error during AC fallback:", e);
+              // Optionally handle other non-NotFound errors from AC fetch.
+            } else {
+              console.debug("Action result not found in AC.");
+            }
+          });
+        }
+
+        if (streamFallback) {
```

Comment:
I think this might be introducing a race condition where both `fetchActionResult` and `streamExecution` try to fetch the action and then set `this.state.actionResult`. We would prefer to use the cached ExecuteResponse or stream-based ExecuteResponse if set, since that gives us the actual response that was returned to bazel for this execution ID.

I think maybe a decent solution here would be only call `fetchActionResult` if `actionDigest` is set in the URL, and `executionId` is _not_ set in the URL.

Reply:
I ironed out the logic here a bit.

The conditions now looks more like this

```
If executeResponseDigest:
  fetchExecuteResponseByDigest()
else if executionId:
  fetchExecuteResponseByActionDigest()
else:
  fetchActionResult()
```

I think it's much cleaner than the previous attempt.

Though it's still a bit unclear to me when `executeResponseDigest` is NOT set and `executionId` is set. Is the logic there mainly for backward compatibility with data created from older app versions?